File: auth-api/src/modules/user/repository/UserRepository.js
import User from "../model/User.js";
import bcrypt from 'bcrypt';

class UserRepository {

  async store(req) {
    try {
      let { name, email, cpf, birthdate, password } = req.body;
      password = await bcrypt.hash(password, 10);
      console.log(password);
      return await User.create({ name, email, cpf, birthdate, password });
    } catch (err) {
      console.error(err.message);
      return null;
    }
  }

  async findById(id) {
    try {
      return await User.findOne({ where: { id } });
    } catch (err) {
      console.error(err.message);
      return null;
    }
  }

  async findByEmail(email) {
    try {
      return await User.findOne({ where: { email } });
    } catch (err) {
      console.error(err.message);
      return null;
    }
  }
}

export default new UserRepository();

File: finance-api/src/main/java/com/taurus/financeapi/modules/user/repository/UserRepository.java
package com.taurus.financeapi.modules.user.repository;

import com.taurus.financeapi.modules.user.dto.UserResetPasswordRequest;
import com.taurus.financeapi.modules.user.dto.UserResponse;
import com.taurus.financeapi.modules.user.model.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Integer> {

    @Query("select new com.taurus.financeapi.modules.user.dto.UserResponse(u.id, u.name, u.cpf, u.birthDate, u.email) from User u where u.email = ?1 and u.password = ?2")
    Optional<UserResponse> getByEmailAndPassword(String email, String password);

    User findByPersonId(String personId);

    @Query(value = "update user u set password = ?1 where u.id = ?2", nativeQuery = true)
    void alterarSenha(UserResetPasswordRequest newPassword);
}